packages/web-components/src/components/ic-typography/ic-typography.tsx

```diff
@@ -20,8 +20,66 @@ export class Typography {
    */
   @Prop() applyVerticalMargins?: boolean = false;
 
+  /**
+   * The number of lines to display before truncating the text, only used for the 'body' variant.
+   */
+  @Prop() maxLines?: number;
+
+  @State() truncationLink: string = "See More...";
+  @State() truncation: boolean;
+  @State() truncStyle: any;
+
+  componentWillLoad(): void {
+    this.maxLines > 0 && this.variant == "body"
+      ? (this.truncStyle = {
+          "max-height": this.boxHeight(this.maxLines),
+          overflow: "hidden",
+        })
+      : null;
+  }
+
+  private boxHeight = (lines: number): any => {
+    return (lines * 1.5).toString() + "rem";
```

Comment:
where does the 1.5 come from? should we have a global variable for this

Reply:
the 1.5 is the global line-height of the body variant to calculate the correct size box, using the design token wouldn't work in this calc I don't think?